compiler/rustc_session/src/session.rs

```diff
@@ -638,6 +638,7 @@ impl Session {
             || self.opts.unstable_opts.unpretty.is_some()
             || self.opts.output_types.contains_key(&OutputType::Mir)
             || std::env::var_os("RUSTC_LOG").is_some()
+            || std::env::var_os("RUSTC_ICE_LOG").is_some()
```

Comment:
@jyn514 identified the issue with defaulting the log level to `TRACE` as likely related to (and worked around by) this bit of logic. I don't know if we will want the default output to start to include SpanTraces or not but I'm leaning towards defaulting them to off for now. We can always bump Span levels up to `WARN` if they're super relevant and we want them to show up all the time.

This way we can leave actually increasing the usage of this functionality to later PRs where it can be based on more practical usage experience.